test/controllers/articleRatingController.test.js

```diff
@@ -0,0 +1,177 @@
+const chai = require('chai'),
+  sinon = require('sinon'),
+  sinonChai = require('sinon-chai'),
+  mockData = require('../mockData'),
+  { Articles, Rating, User } = require('../../database/models'),
+  UserModelManager = require('../../lib/modelManagers/usermodel'),
+  articleRating = require('../../controllers/articleRatingController'),
+  rating = require('../../lib/modelManagers/articleRatingModel'),
+  {
+    OK_CODE,
+    RESOURCE_CREATED_CODE,
+    SERVER_ERROR_MESSAGE,
+    SERVER_ERROR_CODE,
+    NOT_FOUND_CODE,
+    GET_USER_RATING,
+    GET_USER_RATING_ERROR,
+    RATING_CREATED
+  } = require('../../constants/index');
+
+const { expect } = chai;
+
+chai.use(sinonChai);
+
+let newRating = {};
+
+describe('Test Article Rating Controller', () => {
+  const userData = {
+    email: mockData.userdata.email,
+    password: mockData.userdata.password,
+    firstname: mockData.userdata.firstname,
+    lastname: mockData.userdata.lastname
+  };
+
+  before(async () => {
+    const reviewer = await UserModelManager.create(
+      userData.email,
+      userData.password,
+      userData.firstname,
+      userData.lastname
+    );
+
+    const writeUp = await Articles.create(mockData.article);
+
+    newRating = {
+      articleId: writeUp.dataValues.id,
+      reviewerId: reviewer.dataValues.id,
+      rating: 5
+    };
+  });
+
+  afterEach(() => sinon.restore());
+  after('Delete Tables', async () => {
+    Rating.destroy({
+      where: {}
+    });
+
+    User.destroy({
+      where: {}
+    });
+
+    Articles.destroy({
+      where: {}
+    });
+  });
+
+  context('Test controller that creates article rating', () => {
+    it('should send server error if database server is down when creating users rating', async () => {
+      const req = {
+        body: { },
+      };
+      const res = {
+        status() { return this; },
+        json() { }
+      };
+
+      sinon.spy(res, 'status');
+      sinon.spy(res, 'json');
+
+      await articleRating.createRating(req, res);
+      expect(res.status).to.have.been.calledWith(SERVER_ERROR_CODE);
+      expect(res.json.firstCall.lastArg).to.be.an('object');
+      expect(res.json.firstCall.lastArg).to.haveOwnProperty('success').equal(false);
+      expect(res.json.firstCall.lastArg).to.haveOwnProperty('message').equal(SERVER_ERROR_MESSAGE);
+    });
+
+    it('should send information of new rating created', async () => {
+      const req = {
+        body: newRating,
+      };
+      const res = {
+        status() { return this; },
+        json() { }
+      };
+
+      sinon.spy(res, 'status');
+      sinon.spy(res, 'json');
+
+      await articleRating.createRating(req, res);
+      expect(res.status).to.have.been.calledWith(RESOURCE_CREATED_CODE);
+      expect(res.json.firstCall.lastArg).to.be.an('object');
```

Comment:
Adding an assertion to check that the content returned from the server meets expectation would be great. Don't forget to apply this, everywhere necessary.

Reply:
Forgot to do that, I'll do that now